doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,67 @@
+# Troubleshooting Production Issues related to Unsupported Operations in Mountpoint
+
+Mountpoint enables users to access S3 buckets through local filesystem. While this allows for existing workflows and applications to work with S3 unchanged, it also limits the effective API for Mountpoint to be a subset of the filesystem API.
+So, there are a few operations in filesystem API which are not supported using Mountpoint on a mounted bucket. Here, we are enumerating those common cases with their errors reported in the logs which customers face. Please look at [SEMANTICS document](doc/SEMANTICS.md) for more information on what operations are allowed in Mountpoint.
+
+## Non-sequential Write
+Mountpoint only supports sequential writes. On random out of order writes users should expect `Invalid Argument`. For example - 
+```
+$ dd if=/dev/random of=out seek=1 count=1
+dd: writing to 'out': Invalid argument
+```
+In the logs, we get the following WARN message - 
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: mountpoint_s3::fuse: write failed:\
+upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
+
+## Writing to an existing File 
+Mountpoint does support overwriting to existing file using mount option `--allow-overwrite` . Trying to open an existing file for writing without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+In the logs, we get the following WARN message - 
+```
+WARN file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
+```
```

Comment:
I liked the example you gave above with `dd`. Can you have one with something like...:

```sh
echo "Overwriting a file..." > /mnt/s3-bucket/existing-file.txt
```

nit: "Mountpoint supports" instead of "Mountpoint does support"

Reply:
echo gives different error -
`WARN setattr{req=12 ino=2 name="file.txt"}: mountpoint_s3::fuse: setattr failed: inode error: inode 2 (full key "file.txt") is a remote inode and its attributes cannot be modified`

Reply:
I was trying to use vim. But, it is able to open the file using vim. I am just not able to overwrite. Not sure if I should add that as an example.

Reply:
I have updated the error according to my new PR #718 for the case of overwrite and modifying metadata.